cmake/modules/CheckYasm.cmake

```diff
@@ -28,16 +28,31 @@ macro(check_yasm_support _object_format _support_x64 _support_x64_and_avx2)
         if(NOT not_support_avx2)
           set(${_support_x64_and_avx2} TRUE)
         endif(NOT not_support_avx2)
+        execute_process(COMMAND yasm -D HAVE_AS_KNOWS_AVX512 -f ${object_format}
+          -i ${CMAKE_SOURCE_DIR}/src/isa-l/include/
+          ${CMAKE_SOURCE_DIR}/src/isa-l/erasure_code/gf_vect_dot_prod_avx512.asm
+          -o /dev/null
+          RESULT_VARIABLE not_support_avx512
+          OUTPUT_QUIET
+          ERROR_QUIET)
+        if(NOT not_support_avx512)
+          set(${_support_x64_and_avx512} TRUE)
+        endif(NOT not_support_avx512)
       endif(not_arch_x32)
     endif(CMAKE_SYSTEM_PROCESSOR MATCHES "amd64|x86_64")
   endif(NOT no_yasm)
   if(no_yasm)
     message(STATUS "Could NOT find Yasm")
   elseif(NOT not_arch_x32)
       message(STATUS "Found Yasm: only support ILP32 i.e. x86_32")
-  elseif(${_support_x64_and_avx2})
-    message(STATUS "Found Yasm: good -- capable of assembling x86_64 and AVX2")
-  elseif(${_support_x64})
-    message(STATUS "Found Yasm: better -- capable of assembling x86_64")
+  endif()
+  if(${_support_x64_and_avx512})
```

Comment:
might want to replace `_support_x64_and_avx512` with `_support_avx512`.

Reply:
Done